go/teams/member_set.go

```diff
@@ -0,0 +1,135 @@
+package teams
+
+import (
+	"errors"
+
+	"golang.org/x/net/context"
+
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type member struct {
+	version    UserVersion
+	perUserKey keybase1.PerUserKey
+}
+
+type memberSet struct {
+	Owners     []member
+	Admins     []member
+	Writers    []member
+	Readers    []member
+	None       []member
+	recipients map[string]keybase1.PerUserKey
+}
+
+func newMemberSet(ctx context.Context, g *libkb.GlobalContext, req ChangeReq) (*memberSet, error) {
+	set := &memberSet{recipients: make(map[string]keybase1.PerUserKey)}
+	if err := set.loadMembers(ctx, g, req); err != nil {
+		return nil, err
+	}
+	return set, nil
+}
+
+func (m *memberSet) loadMembers(ctx context.Context, g *libkb.GlobalContext, req ChangeReq) error {
+	var err error
+	m.Owners, err = m.loadGroup(ctx, g, req.Owners)
+	if err != nil {
+		return err
+	}
+	m.Admins, err = m.loadGroup(ctx, g, req.Admins)
+	if err != nil {
+		return err
+	}
+	m.Writers, err = m.loadGroup(ctx, g, req.Writers)
+	if err != nil {
+		return err
+	}
+	m.Readers, err = m.loadGroup(ctx, g, req.Readers)
+	if err != nil {
+		return err
+	}
+	m.None, err = m.loadGroup(ctx, g, req.None)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m *memberSet) loadGroup(ctx context.Context, g *libkb.GlobalContext, group []string) ([]member, error) {
+	members := make([]member, len(group))
+	var err error
+	for i, username := range group {
+		members[i], err = m.loadMember(ctx, g, username)
```

Comment:
At the time I started, I don't think UPAK had the necessary fields, but looks like they do now.